bazel/utils/clang_format.bzl

```diff
@@ -0,0 +1,48 @@
+def _clang_format(ctx):
+    clang_format_file = ctx.file.clang_format
+    out_file = ctx.actions.declare_file(ctx.label.name + ".bash")
+    substitutions = {
+        "{pattern-to-format}": ctx.attr.pattern_to_format,
+        "{clang-format}": clang_format_file.short_path,
+        "{style}": ctx.attr.style,
+    }
+    ctx.actions.expand_template(
+        template = ctx.file._template,
+        output = out_file,
+        substitutions = substitutions,
+        is_executable = True,
+    )
+
+    return DefaultInfo(
+        files = depset([out_file]),
+        runfiles = ctx.runfiles(files = [clang_format_file]),
+        executable = out_file,
+    )
+
+clang_format = rule(
+    implementation = _clang_format,
+    attrs = {
+        "style": attr.string(
+            mandatory = True,
+            # From clang-format-7 --help output.
+            values = ["LLVM", "Google", "Chromium", "Mozilla", "WebKit", "file"],
+            doc = "The coding style to apply.",
+        ),
```

Comment:
Sorry I'm expanding the scope for this... but I would strongly recommend this supported passing a Label pointing to a `.clang-format` file. Eg, we have one such file in our `internal` repository. And further... we will necessarily have different styles per directory. Eg, kernel code needs to be formatted differently.

Reply:
I've added an additional label to support multiple `.clang-format` files.